cdnauth/auth.py

```diff
@@ -0,0 +1,119 @@
+import json
+from functools import wraps
+from urllib.parse import urljoin, urlparse
+
+from flask import (
+    Blueprint,
+    current_app,
+    make_response,
+    redirect,
+    request,
+    session,
+    url_for,
+)
+
+from onelogin.saml2.auth import OneLogin_Saml2_Auth
+
+
+def is_safe_url(target):
+    ref_url = urlparse(request.host_url)
+    test_url = urlparse(urljoin(request.host_url, target))
+    return test_url.scheme in ("http", "https") and ref_url.netloc == test_url.netloc
+
+
+def login_required(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if "samlNameId" not in session:
+            return redirect(url_for("auth.saml", sso=True, next=request.url))
+
+        return f(*args, **kwargs)
+
+    return decorated_function
+
+
+def load_saml_settings():
+    json_settings = {}
+    with open("saml/settings.json", "r") as json_file:
+        json_settings = json.load(json_file)
+        json_settings["debug"] = current_app.config["DEBUG"]
+        json_settings["sp"]["entityId"] = current_app.config["SP_ENTITY_ID"]
+        json_settings["sp"]["assertionConsumerService"]["url"] = current_app.config[
+            "SP_ACS_URL"
+        ]
+        json_settings["sp"]["x509cert"] = current_app.config["SP_CERT"]
+        json_settings["sp"]["privateKey"] = current_app.config["SP_KEY"]
+        json_settings["idp"]["entityId"] = current_app.config["IDP_ENTITY_ID"]
+        json_settings["idp"]["singleSignOnService"]["url"] = current_app.config[
+            "IDP_SSO_URL"
+        ]
+        json_settings["idp"]["x509cert"] = current_app.config["IDP_CERT"]
+        json_settings["security"]["wantAssertionsEncrypted"] = current_app.config[
+            "SP_SECURITY_ASSERTIONS_ENCRYPTED"
+        ]
+
+    return json_settings
+
+
+def prepare_flask_request(request):
```

Comment:
Feeling like maybe a docstring here would be helpful.  I had to poke around a bit to understand that the variable `req` which is set from the output of this function (around line 76), is a dictionary in the expected form that the `python3-saml` class `OneLogin_Saml2_Auth` expects for `init`.  

Otherwise, I wondered a bit why we'd collect things from the `request` object that are readily available.